ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostSingleton.java

```diff
@@ -208,27 +265,44 @@ private static void startChartboost(Activity activity, ChartboostParams params,
 
         if (!mIsChartboostInitialized) {
             mIsChartboostInitializing = true;
-            Chartboost.startWithAppId(activity, params.getAppId(), params.getAppSignature());
-
-            Chartboost.setMediation(Chartboost.CBMediation.CBMediationAdMob,
-                    BuildConfig.VERSION_NAME);
-            Chartboost.setLoggingLevel(CBLogging.Level.INTEGRATION);
+            Chartboost.startWithAppId(context, params.getAppId(), params.getAppSignature());
+            Chartboost.setMediation(Chartboost.CBMediation.CBMediationAdMob, BuildConfig.SDK_VERSION, "8.0.1.0");
+            Chartboost.setLoggingLevel(CBLogging.Level.ALL);
             Chartboost.setDelegate(getInstance());
             Chartboost.setAutoCacheAds(false);
-
-            // Chartboost depends on Activity's lifecycle events to initialize its SDK. Chartboost
-            // requires onCreate, onStart and onResume callbacks to initialize its SDK.  By the time
-            // AdMob SDK requests this adapter to initialize the SDK, all three callbacks might have
-            // already been called. So, we call Chartboost's onCreate, onStart and onResume methods
-            // so that the Chartboost SDK will be initialized.
-            Chartboost.onCreate(activity);
-            Chartboost.onStart(activity);
-            Chartboost.onResume(activity);
+            initBanner(context, params, bannerListener);
+            adapterDelegate.didInitialize();
         } else {
+            initBanner(context, params, bannerListener);
             adapterDelegate.didInitialize();
```

Comment:
This method is used for all ad types right? Aren't we loading banners here even if asked for interstitials or rewarded ads?

Reply:
No because of the check for banner listener inside init banner. I will refactor this a bit to make it more clear.